src/Mailer/Ownership/SendOwnershipMailCommandHandler.php

```diff
@@ -0,0 +1,141 @@
+
+    private Mailer $mailer;
+    private MailsSentRepository $mailsSentRepository;
+    private LoggerInterface $logger;
+    private DocumentRepository $organizerRepository;
+    private UserIdentityResolver $identityResolver;
+    private IriGeneratorInterface $organizerIriGenerator;
+    private TwigEnvironment $twig;
+    private OwnershipSearchRepository $ownershipSearchRepository;
+
+    public function __construct(
+        Mailer $mailer,
+        MailsSentRepository $mailsSentRepository,
+        DocumentRepository $organizerRepository,
+        UserIdentityResolver $identityResolver,
+        IriGeneratorInterface $organizerIriGenerator,
+        TwigEnvironment $twig,
+        OwnershipSearchRepository $ownershipSearchRepository,
+        LoggerInterface $logger
+    ) {
+        $this->mailer = $mailer;
+        $this->mailsSentRepository = $mailsSentRepository;
+        $this->logger = $logger;
+        $this->organizerRepository = $organizerRepository;
+        $this->identityResolver = $identityResolver;
+        $this->organizerIriGenerator = $organizerIriGenerator;
+        $this->twig = $twig;
+        $this->ownershipSearchRepository = $ownershipSearchRepository;
+    }
+
+    public function handle($command): void
+    {
+        switch (true) {
+            case $command instanceof SendOwnershipRequestedMail:
+                $this->sentMail(
+                    $command,
+                    self::SUBJECT_OWNERSHIP_REQUESTED,
+                    self::TEMPLATE_OWNERSHIP_REQUESTED
+                );
+                break;
+
+        }
+    }
+
+    public function sentMail(SendOwnershipRequestedMail $command, string $rawSubject, string $template): void
+    {
+        $uuid = new Uuid($command->getUuid());
+
+        if ($this->mailsSentRepository->isMailSent($uuid, get_class($command))) {
+            return;
+        }
+
+        try {
+            $ownershipItem = $this->ownershipSearchRepository->getById($uuid->toString());
+        } catch (OwnershipItemNotFound $e) {
+            $this->logger->warning('[ownership-mail] ' . $e->getMessage());
+            return;
+        }
+
+        try {
+            $organizerProjection = $this->organizerRepository->fetch($ownershipItem->getItemId());
+        } catch (DocumentDoesNotExist $e) {
+            $this->logger->warning(sprintf('[ownership-mail] Could not load organizer: %s', $e->getMessage()));
+            return;
+        }
+
+        $organizer = $organizerProjection->getAssocBody();
+
+        //@todo loop over ALL owners of organisation
+        $ownerId = $ownershipItem->getOwnerId();
+        $ownerDetails = $this->identityResolver->getUserById($ownerId);
+
+        if ($ownerDetails === null) {
+            $this->logger->warning(sprintf('[ownership-mail] Could not load owner details for %s', $ownerId));
+            return;
+        }
+
+        $params = [
+            'organisationName' => $organizer['name']['nl'],
```

Comment:
You can't be sure that the `nl` name of the organizer is provided

Reply:
A while ago you created a class for this to solve that issue for Places `ExtractOfferName`. I guess you could create a more general class that extracts a translated property. So you create a method that gets the data and the name of the property to retrieve.